web/gridt/models/user.py

```diff
@@ -121,6 +123,26 @@ def get_password_reset_token(self):
         token = jwt.encode(token_dict, secret_key, algorithm="HS256").decode("utf-8")
         return token
 
+    def get_email_change_token(self, new_email):
+        """
+        Make a dictionary containing the user's id, new email 
+        + an expiration timestamp such that the token is valid for 2 hours
+        and encodes it into a JWT.
+
+        :param str token: the JWT that is used to verify the e-mail change.
+        """
+        now = datetime.datetime.now()
+        valid = datetime.timedelta(hours=2)
+        exp = now + valid
+        exp = exp.timestamp()
+
+        secret_key = current_app.config["SECRET_KEY"]
+
+        token_dict = {"user_id": self.id, "new_email": new_email, "exp": exp}
+
+        token = jwt.encode(token_dict, secret_key, algorithm="HS256").decode("utf-8")
```

Comment:
I'm sorry, I'm a noob at this. But why first encode and then decode again?

Reply:
I thought I had an answer, then did some research to make sure. Now I don't have an answer anymore. @PJvanVliet ??

Reply:
The "decode" is only to get the token into string form, so it can be put into a url.